snap/validate.go

```diff
@@ -272,7 +272,7 @@ func validateSocketAddrPath(socket *SocketInfo, fieldName string, path string) e
 		return fmt.Errorf("invalid %q: %q should be written as %q", fieldName, path, clean)
 	}
 
-	if !(strings.HasPrefix(path, "$SNAP_DATA/") || strings.HasPrefix(path, "$SNAP_COMMON/")) {
+	if !(strings.HasPrefix(path, "$SNAP_DATA/") || strings.HasPrefix(path, "$SNAP_COMMON/") || strings.HasPrefix(path, "/run/user/0/$SNAP_NAME/")) {
 		return fmt.Errorf(
 			"invalid %q: must have a prefix of $SNAP_DATA or $SNAP_COMMON", fieldName)
```

Comment:
If we're going this way, the message needs to be  updated to include `/run/user/0/...`

Reply:
I agree with @bboozzoo on this point.

Reply:
I did updated message, though with XDG_RUNTIME_DIR as @jdstrand proposed